docs/DeveloperGuide.md

```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
+
+
+**Aspect 2: What should be the inputs for GradedTest Constructor:**
+
+**Alternative 1:** Using Strings for Graded Test Constructor.
+- Pros:
+  * Versatility: Utilizing strings allows for versatile inputs, accommodating various formats and representations.
+  * Customization: Users can input graded test scores in a way that suits their preferences, including default values or special symbols.
+- Cons:
+  * Parsing Overhead: The need for additional parsing introduces complexity, especially when validating inputs.
+  * Type Safety: The reliance on strings may lead to potential type-related issues during runtime.
+  * Complexity: Since all the scores are in a single string, additional parsing will be needed to implement getters and setters.
+  * Learning Curve: Users might need to adapt to a more structured input format, potentially increasing the learning curve.
+  
+**Alternative 2:** Using Structured Object for Graded Test Constructors.
+- Pros:
+  * Type Safety: A structured object ensures type safety, reducing the risk of runtime errors related to data types.
+  * Clarity and Readability: A well-defined object provides clarity and improves the readability of the code, making it easier to understand.
+- Cons:
+  * Less Flexibility: All 5 fields, `ra1, ra2, midterms, finals, pe` needs to be present at all times. Need to create 5 objects for 1 graded test.
+  * Logging/Documentation: Additional parsing is needed to properly store these fields in the json file properly.
+
+**Alternative 3 (current choice):** Use both Strings and Structured Objects for Graded Test Calculators.
+- Pros:
+  * Type Safety Utilization: Ensures type safety with objects and allows the use of strings for flexibility. We can take advantage of the strings to enable the use of `default` values.
+  * Robust Testing: Better bug identification, as the two constructors check each other.
+- Cons:
+  * Extra caution is needed to ensure that both these constructors are compatible with one another. 
+
+
+**Aspect 3: How to store graded test scores for individuals**
+
+**Alternative 1(current choice):** Piggy back on the Person Object
+- Pros:
+  * Coherence: Graded test information is closely associated with the person(student) it belongs to.
+  * Simplicity: Simplifies the overall structure by leveraging the existing Person object.
+- Cons:
+  * Immutability: Any updates to graded test scores to the Person Objects may cause issues down stream on methods referencing to the old Person object.
+  * Complex Handling of Person: More logic will have to be implemented to update the references of the new Person object.
+
+**Alternative 2:** Isolate the GradedTest Object as its own 
+- Pros:
+  * Easier to debug: Since the GradedTest Object is isolated.
+- Cons:
+  * Immutability: Any updates to Person Object will result in the GradedTest Object referencing the deleted person.
+
+
+**Aspect 4: How to initialise a graded test instance for users**
```

Comment:
Same as for Aspect 3, should end with a colon 

OR alternatively remove the colon at the end of the lines for Aspects 1 and 2 for standardisation